### PR TITLE
Update Dompdf library to version 0.8.2

### PR DESCRIPTION
**Pull request by knazart**
The Dompdf library was updated to 0.8.2

## How has this been tested
* [x] Try generating a PDF in any way possible. For example, try downloading a PDF of the order receipt after a successful event registration. There should be no PHP Warnings or Deprecated messages in the logs.
* [x] Try using custom templates.
* [x] Test generating an invoice PDF while the "Enfold" theme is active on the site.
* [x] Tests the eea-utf8-message-template-variations add-on with this PR, the **TASK/add-new-cache-file-used-in-dompdf-0.8** branch if not yet merged.
(https://github.com/eventespresso/eea-utf8-message-template-variations/pull/1)

## Checklist

* [x] I have added a changelog entry for this pull request
* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [x] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.
* [x] My code accounts for when the site is Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

**Comment by mnelson4** [comment]
note: I believe this is the ticket from the last time we updated dompdf: https://events.codebasehq.com/projects/event-espresso/tickets/6804
I tried using the old utf-8 template variation (https://github.com/eventespresso/eea-utf8-message-template-variations), but couldn't get it to work with master or this branch

**Comment by mnelson4** [comment]
Also, we should retest generating an invoice PDF while the "Enfold" theme is active on the site, as that was an issue last time we updated dompdf. See https://events.codebasehq.com/projects/event-espresso/tickets/7132

File: core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php [hunk 1]
```diff
@@ -1,72 +1,67 @@
...
-  'fixed' =>
-  array (
-    'normal' => DOMPDF_FONT_DIR . 'Courier',
-    'bold' => DOMPDF_FONT_DIR . 'Courier-Bold',
-    'italic' => DOMPDF_FONT_DIR . 'Courier-Oblique',
-    'bold_italic' => DOMPDF_FONT_DIR . 'Courier-BoldOblique',
-  )
-) ?>
+<?php
+$distFontDir = $rootDir . DIRECTORY_SEPARATOR . 'lib' . DIRECTORY_SEPARATOR . 'fonts' . DIRECTORY_SEPARATOR;
```
Comment (mnelson4) [inline core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: we should probably surround this in a check for `is_defined('DOMPDF_FONT_DIR')` as we still allow folks to put their invoice's fonts in uploads (see `core/espresso_definitions.php`)

File: core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php [hunk 2]
```diff
@@ -1,72 +1,67 @@
...
+            'bold_italic' => $distFontDir . 'Courier-BoldOblique'
+        ),
+    'fixed' =>
+        array(
+            'normal' => $distFontDir . 'Courier',
+            'bold' => $distFontDir . 'Courier-Bold',
+            'italic' => $distFontDir . 'Courier-Oblique',
+            'bold_italic' => $distFontDir . 'Courier-BoldOblique'
+        )
+);
```
Comment (mnelson4) [inline core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: we used to have zpfdingbats in here. We should probably re-add it, as I think it helped with emojis or something

File: tests/mocks/addons/new-payment-method/.gitignore [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # auto-generated/build-artifact files that should not be committed
 core/third_party_libs/dompdf/lib/fonts/log.htm
+core/third_party_libs/dompdf/lib/fonts/*.afm.php
+core/third_party_libs/dompdf/lib/fonts/*.ufm.php
```
Comment (mnelson4) [inline tests/mocks/addons/new-payment-method/.gitignore]: these lines in `.gitignore` were actually in there by accident- the `new-payment-method` add-on doesn't have a `core/third_party_libs/dompdf/lib/fonts` directory anyway. So these lines may as well be removed from all the skeleton add-ons' `.gitignore` files.

File: core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php (hunk 1 above)
Comment (knazart) [reply core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: thanks, I did miss the scenario of fonts in uploads.

**Comment by knazart** [comment]
Ok, I don't really have access to the "Enfold" theme and the link Josh provided in Ticket 7132 is not working anymore. Might want to just ask him to test this.

> I tried using the old utf-8 template variation , but couldn't get it to work with master or this branch.

If it wan't working before, do we have an issue open for that ? Or do you want to try and do a fix in this PR ? Sorry, I just don't follow.

File: core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php (hunk 2 above)
Comment (mnelson4) [reply core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: ah ok, ya that's right, we removed zapdingbats font files... we just erroneously didn't remove it from the font cache file. So you're right to remove it from both places then :+1:

Comment (knazart) [reply core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: So we don't want it after all huh ? Revert the `add zapfdinbats back to the core fonts list` back ?

**Comment by mnelson4** [comment]
> Ok, I don't really have access to the "Enfold" theme and the link Josh provided in Ticket 7132 is not working anymore. Might want to just ask him to test this.

Yeah I meant that should just be part of the testing steps.

> If (UTF8 add-on) wasn't working before, do we have an issue open for that ? Or do you want to try and do a fix in this PR ? 

Ideally it should be working, but if it's not working with master then it should be a different ticket... although in the course of thinking about this, I realized what I was doing wrong: I had emojis in the PDF, but the UTF8 add-on never supported those anyways, it just supported a bunch of accented and foreign characters. I saw it IS working with master for that, but not working in this branch...
I found this branch needed a few tweaks to work with that, and unfortunately that add-on will need an update too.
Also, in the process of testing, I noticed my suggestion on https://github.com/eventespresso/event-espresso-core/pull/699/commits/f8c775ec757896f322c0e24cad5f2cf2a13b63f1 was actually unnecessary- if we've told DOMPDF to use a different font folder, it needs to use the cache file from THAT folder too anyway.

So please review my latest tweaks to this branch and the related PR in the UTF8 add-on: https://github.com/eventespresso/eea-utf8-message-template-variations/pull/1. Testing should include double-checking that (updated) add-on works with this branch.

File: core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php (hunk 2 above)
Comment (mnelson4) [reply core/third_party_libs/dompdf/lib/fonts/dompdf_font_family_cache.dist.php]: oh yeah, well we don't have the zapfdinbats font files anyway. So yes I suppose that commit should be reverted

**Comment by knazart** [comment]
If that is ok, assigning to you Josh for testing. The 3rd item in the tests list requires the Enfold theme that you originally had access to, that is if you still do.

**Comment by joshfeck** [comment]
This is ready to be merged.

**Comment by knazart** [comment]
Found another bug (2 last commits).

**Pull request merged**